server/server.rs

```diff
@@ -45,13 +46,14 @@ pub struct Server {
     database_diagnostics_updater: IntervalRunner,
     user_manager: Arc<UserManager>,
     authenticator_cache: Arc<AuthenticatorCache>,
-    typedb_service: Option<TypeDBService>,
+    typedb_service: TypeDBService,
```

Comment:
I was wondering why this was ever `Option<_>` and am coming to the conclusion that the `Server` struct is kinda pointless. It's constructed in `create()` and then _immediately_ deconstructed in `serve()`. The only way to use it as a struct is to access the `database_manager`.

This is out of scope for this PR, but a note for future reference.